src/components/Modal.js

```diff
@@ -0,0 +1,35 @@
+import React, { PureComponent } from "react";
+import "./Modal.css";
+
+class Modal extends PureComponent {
+  ref = React.createRef();
+
+  onClose = () => {
+    // this.ref.current.style = { display: "none" };
+    this.props.onClose();
+  };
+
+  onClick = e => {
+    if (e.target === this.ref.current) this.onClose();
+  };
+
+  render() {
+    const containerStyle = {
+      padding: "25px",
+      display: "flex",
+      flexDirection: "column"
+    };
+    return (
+      <div onClick={this.onClick} ref={this.ref} className="modal">
```

Comment:
Rename this class to `modal-overlay`, I was confused at first, I didn't understand why clicking the modal results with closing.